### PR TITLE
feature/g3t-update

### PR DESCRIPTION
**Pull request by bwalsh**
update for latest g3t commands

### CODE REVIEW

File: docs/requirements/gen3-tracker.md [hunk 1]
```diff
@@ -54,11 +54,11 @@ username: someone@example.com
 
 The following options and environmental variables are synonymous, you may set them as environmental variables or pass them as parameters to the command line.
```
Comment (quinnwai) [inline docs/requirements/gen3-tracker.md:55]: "**make sure to** set them" as something that needs to be done. Also, I think we should mention that we carry on with the rest of the quickstart as if we have chosen to set environmental variables

File: docs/workflows/quick-start-guide.md [hunk 2]
```diff
@@ -1,52 +1,131 @@
...
 
-Add file(s)
 ```
 g3t add folder/file.tsv
 g3t add folder/file2.tsv
 ```
 
-Create meta data for your files. This `g3t` command will take care of creating this for you
+* This command will record an entry for each file in the `MANIFEST` directory.
+* Optionally you can tag each file with patient, specimen or other identifiers. See `g3t add --help` for more information.
```
Comment (quinnwai) [inline docs/workflows/quick-start-guide.md:53]: Shall we link to [/workflows/upload/](https://aced-idp.github.io/workflows/upload/) so they can stay in the docs as opposed to needing to go into the CLI?

File: docs/workflows/quick-start-guide.md [hunk 3]
```diff
@@ -1,52 +1,131 @@
...
 ```
 
-Push to ACED IDP
+### Push to ACED IDP
 ```
 g3t push
 ```
 
-View the upload status (pending or complete)
+Notes:
```
Comment (quinnwai) [inline docs/workflows/quick-start-guide.md:102]: Bullet points below not getting formatted, might just need to add a new line here

File: docs/workflows/quick-start-guide.md [hunk 4]
```diff
@@ -1,52 +1,131 @@
...
+
 ## 1. Upload Data (To an Existing Project on ACED IDP)
-This section outlines how to add data to ACED IDP. Many `g3t` commands behave similarily to `git` commands
+This section outlines how to add data to ACED IDP. Many `g3t` commands behave similarly to `git` commands
+
+### Initialize a new project:
 
-Initialize a new project
 ```
 g3t init aced-demo
```
Comment (quinnwai) [inline docs/workflows/quick-start-guide.md:36]: even after setting `G3T_PROFILE=local`, I get
```
WARNING: No --profile specified, found multiple gen3_client profiles: ['local', 'dev', 'staging', 'cbds-prod', 'cbds-dev']
No profile set. Continuing in disconnected mode. Use `set profile <profile>`
Initialized empty Git repository in /Users/wongq/data/cbds-demo/local/cbds-for_docs/.git/
```

should we make it `g3t --profile <profile_name> init aced-demo`?

File: docs/workflows/quick-start-guide.md [hunk 5]
```diff
@@ -1,52 +1,131 @@
...
 ```
 
-Check that the expected files are queued for upload
+* This command will create metadata files in the META directory. At a minimum, the directory will contain:
+
+| file                     | contents                   |
+|--------------------------|----------------------------|
+| ResearchStudy.ndjson     | Description of the project |
+| DocumentReference.ndjson | File information           |
+
```
Comment (quinnwai) [inline docs/workflows/quick-start-guide.md:68]: Oooh very nice, didn't realize you could do tables like this

File: docs/workflows/quick-start-guide.md [hunk 6]
```diff
@@ -1,52 +1,131 @@
...
+| Specimen.ndjson        | Sample information     |
+
+
+> Note: You may supply your own FHIR data, the `g3t meta init` command is a convenience method to generate the required files.
+
+### Check that the metadata is valid:
+
+```
+g3t meta validate
+```
```
Comment (quinnwai) [inline docs/workflows/quick-start-guide.md:84]: Something like "If the metadata is valid, a set of summary counts will be displayed with no error messages"... might be a little heavy handed

(PS: added an [issue](https://github.com/ACED-IDP/gen3_util/issues/102) here so we can make this clearer on the g3t side not just docs)

File: docs/workflows/quick-start-guide.md [hunk 7]
```diff
@@ -1,52 +1,131 @@
...
 Check that your data was uploaded successfully by viewing on the Exploration tab on [aced-idp.org](https://aced-idp.org)
 
-## 2. Download Data
+## 2. Cloning a project
+
+```commandline
+g3t clone [OPTIONS] PROJECT_ID 
+```
+
+The clone command will download the MANIFEST/ and META/ directories associated with the project.
```
Comment (quinnwai) [inline docs/workflows/quick-start-guide.md:121]: maybe...
> associated with the project _into its own directory_

Comment (quinnwai) [reply docs/workflows/quick-start-guide.md:121]: also, I'm not able to use env var G3T_PROFILE here as well, had to se a profile.
```
WARNING: No --profile specified, found multiple gen3_client profiles: ['local', 'dev', 'staging', 'cbds-prod', 'cbds-dev']
INFO: No config file found in current directory or parents.
'NoneType' object has no attribute 'strip'
```

Shall we include --profile flag in the example command?

File: mkdocs.yml [hunk 8]
```diff
@@ -8,11 +8,11 @@ site_name: ACED-IDP Documentation
...
-- workflows/quick-start-guide.md
-- Requirements:
+#- getting-started.md
+- Getting Started:
   - requirements/gen3-client.md
   - requirements/gen3-tracker.md
+  - workflows/quick-start-guide.md
 - Use Cases & Workflows:
   - workflows/creating-project.md
   - Submitting Data:
```
Comment (quinnwai) [inline mkdocs.yml:18]: On some other pages..
- mention of deprecated g3t utilities in `workflows/restrict-access/` url
- I think there's some outdated info on adding metadata (eg [/workflows/metadata](https://aced-idp.github.io/workflows/metadata/)) where Task and Diagnostic Report are not currently supported in our submission ETL

File: docs/workflows/quick-start-guide.md [hunk 9]
```diff
@@ -1,52 +1,131 @@
...
+Alternatively, you can set the environmental variables using the `EXPORT` function e.g.:
+
+```sh
+export G3T_PROJECT_ID=aced-myproject
+```
+
+
 ## 1. Upload Data (To an Existing Project on ACED IDP)
-This section outlines how to add data to ACED IDP. Many `g3t` commands behave similarily to `git` commands
+This section outlines how to add data to ACED IDP. Many `g3t` commands behave similarly to `git` commands
```
Comment (quinnwai) [inline docs/workflows/quick-start-guide.md:31]: Add something about how users need to be granted permissions to a project on ACED? (hence explaining why it has to be an existing project?)

(might have missed this being mentioned somewhere else)

File: mkdocs.yml [hunk 10]
```diff
@@ -8,11 +8,11 @@ site_name: ACED-IDP Documentation
 
 nav:
 - index.md
-- getting-started.md
-- workflows/quick-start-guide.md
-- Requirements:
+#- getting-started.md
+- Getting Started:
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply mkdocs.yml:12]: done

File: docs/requirements/gen3-tracker.md [hunk 11]
```diff
@@ -54,11 +54,11 @@ username: someone@example.com
...
-| --project_id | G3T_PROJECT_ID  | authorization       |                                |
-| --profile    | G3T_PROFILE     | gen3-client profile |                                |
-| --format     | G3T_FORMAT      | Output format.      | yaml         |
+| option       | environment     | comment             | example       |
+|--------------|-----------------| ------------------- |---------------|
+| --project_id | G3T_PROJECT_ID  | authorization       | aced-test     |
+| --profile    | G3T_PROFILE     | gen3-client profile | production    |
+| --format     | G3T_FORMAT      | Output format.      | default: yaml |
 
 Alternatively, you can set the environmental variables using the `EXPORT` function e.g.:
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply docs/requirements/gen3-tracker.md:63]: done

File: docs/requirements/gen3-tracker.md [hunk 12]
```diff
@@ -54,11 +54,11 @@ username: someone@example.com
...
 The following options and environmental variables are synonymous, you may set them as environmental variables or pass them as parameters to the command line.
 
-| option       | environment     | comment             | example                        |
-|--------------|-----------------| ------------------- | ------------------------------ |
-| --project_id | G3T_PROJECT_ID  | authorization       |                                |
-| --profile    | G3T_PROFILE     | gen3-client profile |                                |
-| --format     | G3T_FORMAT      | Output format.      | yaml         |
+| option       | environment     | comment             | example       |
+|--------------|-----------------| ------------------- |---------------|
+| --project_id | G3T_PROJECT_ID  | authorization       | aced-test     |
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply docs/requirements/gen3-tracker.md:59]: done

File: docs/workflows/quick-start-guide.md [hunk 13]
```diff
@@ -1,52 +1,131 @@
...
+### Add file(s):
 
-Add file(s)
 ```
 g3t add folder/file.tsv
 g3t add folder/file2.tsv
 ```
 
-Create meta data for your files. This `g3t` command will take care of creating this for you
+* This command will record an entry for each file in the `MANIFEST` directory.
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:52]: done

File: docs/workflows/quick-start-guide.md [hunk 14]
```diff
@@ -1,52 +1,131 @@
...
 ```
 
-Clone an existing project
-```
-g3t clone aced-demo
-```
+* Similar to `git init`, this command initializes a new project in the current directory
+* The id is `aced-demo`; the program name is `aced`, the project name is `demo`
+* `Program name` is predefined by the institution, the `g3t ping` command will show the available programs
+* `Project name` must be unique within the server, and must be alphanumeric and contain no spaces
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:42]: done

File: docs/workflows/quick-start-guide.md (hunk 6 above)
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:84]: done

File: docs/workflows/quick-start-guide.md (hunk 7 above)
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:121]: Works for me: ❓ 

```$ G3T_PROFILE=local g3t clone cbds-testgrip
Cloning -[2024-10-07 18:33:05,425][   INFO] File .g3t/work/cbds-testgrip.git.zip downloaded successfully
Cloned .g3t/work/cbds-testgrip.git.zip
On branch main
nothing to commit, working tree clean
```

File: docs/workflows/quick-start-guide.md (hunk 9 above)
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:31]: mentioned somewhere else

File: mkdocs.yml (hunk 8 above)
Comment (bwalsh) [reply mkdocs.yml:18]: DIagnosticReport, restrict-access: `deprecated`
done

File: docs/workflows/quick-start-guide.md [hunk 15]
```diff
@@ -1,52 +1,134 @@
...
+Please ensure you have the following dependencies installed:
+
+* [gen3-client](../requirements/gen3-client.md)
+* [gen3-tracker](../requirements/gen3-tracker.md)
+
+## Usage
+
+`g3t  [OPTIONS] COMMAND [ARGS]...`
+
+The following options and environmental variables are synonymous, you may set them as environmental variables or pass them as parameters to the command line.
```
Comment (rlancaster96) [inline docs/workflows/quick-start-guide.md:15]: I haven't been able to figure out how to successfully pass the profile name on the command line. In my current workflow I set env variables G3T_PROFILE and G3T_PROJECT. Could you add an example of running a g3t command with options, command, and args that includes setting profile?
```
(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ gen3-client configure --profile=cbds --cred=/home/groups/CEDAR/lancasru/.gen3/credentials.json --apiendpoint=https://idp.cbds.ohsu.edu
2024/10/09 10:37:20 A new version of gen3-client is available! The latest version is 2024.10.0. You are using version 2024.09
2024/10/09 10:37:20 Please download the latest gen3-client release from https://github.com/uc-cdis/cdis-data-client/releases/latest
2024/10/09 10:37:20 Profile 'cbds' has been configured successfully.
(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ g3t ping
WARNING: No --profile specified, found multiple gen3_client profiles: ['cedar', 'cbds']
msg: 'Configuration ERROR: No profile found'

(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ export G3T_PROFILE=cbds
(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ g3t ping
msg: 'Configuration OK: Connected using profile:cbds'
endpoint: https://idp.cbds.ohsu.edu
username: lancasru@ohsu.edu

(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ export G3T_PROFILE=""
(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ g3t ping
WARNING: No --profile specified, found multiple gen3_client profiles: ['cedar', 'cbds']
msg: 'Configuration ERROR: No profile found'

(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ g3t ping --profile 'cbds'
WARNING: No --profile specified, found multiple gen3_client profiles: ['cedar', 'cbds']
Usage: g3t ping [OPTIONS]
Try 'g3t ping --help' for help.

Error: No such option: --profile
(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ g3t ping --profile cbds
WARNING: No --profile specified, found multiple gen3_client profiles: ['cedar', 'cbds']
Usage: g3t ping [OPTIONS]
Try 'g3t ping --help' for help.

Error: No such option: --profile
(g3t_venv) (python311) [lancasru@arc-infra-3 NHW]$ g3t ping cbds
WARNING: No --profile specified, found multiple gen3_client profiles: ['cedar', 'cbds']
Usage: g3t ping [OPTIONS]
Try 'g3t ping --help' for help.

Error: Got unexpected extra argument (cbds)
```

File: docs/workflows/approve-requests.md [hunk 16]
```diff
@@ -32,19 +16,38 @@ g3t utilities access sign
...
 ## Use case
 
+# Adding Data Access Committee members
+
+Note: This example uses the ohsu program, but the same process applies to all programs.
+
+* A sysadmin will add the requestor role to a data-access-committee member(s) aka data steward
+* Only users with the requestor role can APPROVE and SIGN adding policies within their program
+
+```text
```
Comment (rlancaster96) [inline docs/workflows/approve-requests.md:41]: Formatting issue with ```; ## Use case and ## Adding Data Access Committee members displays as code.

**Comment by elisabethgoldman** [comment]
> update for latest g3t commands

#### Some feedback on documentation:

1. Under the top heading “Quick Start Guide” page, a few sentences describing the purpose of this documentation, i.e., what task it guides a user through, would be helpful. Something along the lines of “This documentation will walk you through the steps of setting up a project so you are able to upload/download, and keep track of your data using the ACED-IDP.” 
2. Under “Dependencies” heading, maybe add the following to the existing text:
 “Please ensure you have the following dependencies installed. You will need to configure your user profile using the gen3-client in order to use proceed with using g3t. You can use g3t ping to verify your profile is successfully configured.”

3. Under “Usage” heading, a sentence describing/introducing g3t would also be helpful, instead of having this statement “Many g3t commands behave similarly to git commands” under “Upload Data”, move to under Usage heading so it says something like: “g3t is the command line tool used to interact with the ACED-IDP. Many g3t commands behave similarly to git commands”
4. I would add a comment under “Upload Data” about when one should use g3t init vs g3t clone.
5. The way the following is written is confusing because of the reference to both an existing and a ‘new’ project:

## 1. Upload Data (To an Existing Project on ACED IDP)

This section outlines how to add data to ACED IDP.

### Initialize a new project:

`g3t init aced-myproject`

*Could change to something like the following to make more clear:* 

## 1. Uploading Data

This section outlines how to add data to an existing project on ACED-IDP with the project name `myproject`. The Project ID for this example case is `aced-myproject`, and the program is `aced`. If you wish to create a new project, click on creating a project link for more information. 

### Initialize the project in your terminal:

`g3t init aced-myproject`

### Other recommendations:
6. Note somewhere that wildcards cannot be used to `g3t add` multiple files at once at the present time. 

7. Perhaps add a troubleshooting heading for the following error (not sure how commonly it will be encountered by other users):

 - if you try to `g3t add `> `g3t commit `>` g3t push` files that you initially added but for whatever reason couldn't commit after first adding them, you will get this error message: 

```
Please correct issues before pushing.
Command `g3t status` failed with error code 1, stderr: WARNING: DocumentReference.ndjson is out of date 1970-01-01T01:00:00. The most recently changed file is MANIFEST/DocumentReference.ndjson.dvc 2024-09-29T10:34:58.  Please check DocumentReferences.ndjson                  
No data file changes.
```

To fix this issue, you can use:
`git add META && git commit -m "add META directory" `

File: docs/requirements/gen3-tracker.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (quinnwai) [reply docs/requirements/gen3-tracker.md:59]: one more to change from `aced-test` to `aced-myproject` on the `requirements/gen3-tracker/` page

File: docs/requirements/gen3-tracker.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (quinnwai) [reply docs/requirements/gen3-tracker.md:63]: has this been pushed? Still seeing it unchanged on the [reviewable website](https://deploy-preview-23--luminous-boba-a6e270.netlify.app/requirements/gen3-tracker/#:~:text=The%20command%20g3t%20ping%20will%20confirm%20that%20the%20access%20key%20and%20gen3%2Dclient%20have%20been%20configured%20correctly)

File: docs/workflows/quick-start-guide.md (hunk 15 above)
Comment (matthewpeterkort) [reply docs/workflows/quick-start-guide.md:15]: The --profile option goes after g3t but before the first command. Ex: `g3t --profile local ping` and `g3t --profile local init ohsu-test`

Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:15]: See [adds example](https://github.com/ACED-IDP/aced-idp.github.io/pull/23/commits/ba970c742d750af18d3ca89a106fb31d390905f6)

File: docs/workflows/approve-requests.md (hunk 16 above)
Comment (bwalsh) [reply docs/workflows/approve-requests.md:41]: done.
see "fix formatting" commit

File: mkdocs.yml (hunk 8 above)
Comment (quinnwai) [reply mkdocs.yml:18]: Great thanks! One more thing...

> Additional resources can be added to the metadata files:
> - subjects (ResearchSubject, Patient)
> - specimens (Specimen)
> - assays (Task)
> - measurements (Observation)

 To my understanding, Task isn't being created or used in our current g3t setup. Assay info in SMMART for instance, is stored in the Observation with Specimen reference. Consider maybe condensing last two bullet points into one, eg
> - assays/measurements (Observations)
?

File: docs/workflows/quick-start-guide.md (hunk 7 above)
Comment (quinnwai) [reply docs/workflows/quick-start-guide.md]: forgot to export, my bad

**Review by quinnwai — APPROVED** [review]
A few little nitpicks comments on this thread and some that only show up in the Github [files section](https://github.com/ACED-IDP/aced-idp.github.io/pull/23/files). Only main thing is that the [/getting-started](https://deploy-preview-23--luminous-boba-a6e270.netlify.app/getting-started/) page is orphaned atm. Could we change it to an /overview page within the "Getting Started" section and link to that on the welcome page? That way people can proceeding chronologically through the top left table of contents as opposed to accidentally skipping to the quickstart guide and skipping over the gen3-client and gen3-tracker.

File: docs/workflows/quick-start-guide.md (hunk 9 above)
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:33]: https://deploy-preview-23--luminous-boba-a6e270.netlify.app/workflows/add-users/

File: docs/workflows/quick-start-guide.md (hunk 2 above)
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:55]: done

File: docs/workflows/quick-start-guide.md [hunk 17]
```diff
@@ -1,52 +1,131 @@
...
 g3t add folder/file.tsv
 g3t add folder/file2.tsv
 ```
 
-Create meta data for your files. This `g3t` command will take care of creating this for you
+* This command will record an entry for each file in the `MANIFEST` directory.
+* Optionally you can tag each file with patient, specimen or other identifiers. See `g3t add --help` for more information.
+
+
+### Create meta data for your files. The `g3t meta init` command will take care of creating FHIR compliant data for you.
```
[the remark this answers is not in the document]
Comment (bwalsh) [reply docs/workflows/quick-start-guide.md:59]: done

File: mkdocs.yml (hunk 8 above)
Comment (bwalsh) [reply mkdocs.yml:18]:
```$ g3t add --help | grep task
  --task <value>: The task identifier, requires either patient or specimen
```

**Review by lbeckman314 — APPROVED** [review]

**Comment by quinnwai** [comment]
Hey @rlancaster96 saw you had 2 points that needed addressing, let me know if the dev team addressed them as expected:
- passing in profile as a flag doesn't work, there should be an example:
  -  from Matthew: > The --profile option goes after g3t but before the first command. Ex: `g3t --profile local ping` and `g3t --profile local init ohsu-test`
  - added an example to the docs [here](https://deploy-preview-23--luminous-boba-a6e270.netlify.app/workflows/quick-start-guide/#:~:text=For%20example%2C%20a%20ping%20command%20using%20the%20aced%20profile%20would%20be%20g3t%20%2D%2Dprofile%20aced%20ping)
- [RBAC page](https://deploy-preview-23--luminous-boba-a6e270.netlify.app/workflows/approve-requests/) needed formatting fixes

Let me know if that addresses your points for now! We are planning to do a larger run-through with users, so no worries if you have further points down the road

**Comment by quinnwai** [comment]
Hey @elisabethgoldman, thanks for the feedback! Wanted to make sure your points above were addressed.

------

> 1. Under the top heading “Quick Start Guide” page, a few sentences describing the purpose of this documentation, i.e., what task it guides a user through, would be helpful. Something along the lines of “This documentation will walk you through the steps of setting up a project so you are able to upload/download, and keep track of your data using the ACED-IDP.”

The [welcome page](https://deploy-preview-23--luminous-boba-a6e270.netlify.app/) should have an "About" section detailing this and the [quick start guide](https://deploy-preview-23--luminous-boba-a6e270.netlify.app/workflows/quick-start-guide/) also has a brief description of which use cases will be covered now!

> 2. Under “Dependencies” heading, maybe add the following to the existing text:
“Please ensure you have the following dependencies installed. You will need to configure your user profile using the gen3-client in order to use proceed with using g3t. You can use g3t ping to verify your profile is successfully configured.”

Updated the requirements section within the Quick Start Guide!

> 3. Under “Usage” heading, a sentence describing/introducing g3t would also be helpful, instead of having this statement “Many g3t commands behave similarly to git commands” under “Upload Data”, move to under Usage heading so it says something like: “g3t is the command line tool used to interact with the ACED-IDP. Many g3t commands behave similarly to git commands”

Some info was added to the General Usage heading of the Quick Start Guide, but I think you covered two larger problems that we might address in a further PR:
 1. When to use profile and how g3t "looks" for a profile when you don't have one specified (ie in an init'ed project directory)
 2. How g3t compares to git vs how it is different

Might kick the can down the road on this one since it has a bit of a bigger scope!


> 4. I would add a comment under “Upload Data” about when one should use g3t init vs g3t clone.

Change the last section on the Quick Start to "2. Download Data from a Project on ACED-IDP" and reworded to split out the different ways to download data from a project

> 5. The way the following is written is confusing because of the reference to both an existing and a ‘new’ project:

Changed it to approved project instead to mention that it's existing (ie created and approved) but still new (no data in it)

> 6. Note somewhere that wildcards cannot be used to g3t add multiple files at once at the present time.

Should be addressed!

> 7. Perhaps add a troubleshooting heading for the following error (not sure how commonly it will be encountered by other users):
> - if you try to g3t add > g3t commit > g3t push files that you initially added but for whatever reason couldn't commit after first adding them, you will get this error message:


This relates to the 3. point, I think we'll have to figure out when to use git (if at all) vs g3t so I'll leave that for a separate ticket if that works for you!

------


Let me know if I addressed your feedback!

FYI, we are planning to do a larger run-through with users and rebranding of ACED to CALIPER. So if you have additional points you wanna cover down the road, we'll prolly cover them soon with an in-depth user review

**Pull request merged**